diagram-viewer/src/main/java/com/powsybl/diagram/viewer/MainViewController.java

```diff
@@ -43,6 +43,75 @@
  */
 public class MainViewController {
 
+    public enum ComponentFilterType {
+        ALL,
+        HVDC_LINE,
+        BUS,
+        SWITCH,
+        BUSBAR_SECTION,
+        LINE,
+        TIE_LINE,
+        TWO_WINDINGS_TRANSFORMER,
+        THREE_WINDINGS_TRANSFORMER,
+        GENERATOR,
+        BATTERY,
+        LOAD,
+        SHUNT_COMPENSATOR,
+        DANGLING_LINE,
+        STATIC_VAR_COMPENSATOR,
+        LCC_CONVERTER_STATION,
+        VSC_CONVERTER_STATION,
+        GROUND;
+
+        IdentifiableType toIidm() {
+            return switch (this) {
+                case ALL -> null;
+                case HVDC_LINE -> IdentifiableType.HVDC_LINE;
+                case BUS -> IdentifiableType.BUS;
+                case SWITCH -> IdentifiableType.SWITCH;
+                case BUSBAR_SECTION -> IdentifiableType.BUSBAR_SECTION;
+                case LINE -> IdentifiableType.LINE;
+                case TIE_LINE -> IdentifiableType.TIE_LINE;
+                case TWO_WINDINGS_TRANSFORMER -> IdentifiableType.TWO_WINDINGS_TRANSFORMER;
+                case THREE_WINDINGS_TRANSFORMER -> IdentifiableType.THREE_WINDINGS_TRANSFORMER;
+                case GENERATOR -> IdentifiableType.GENERATOR;
+                case BATTERY -> IdentifiableType.BATTERY;
+                case LOAD -> IdentifiableType.LOAD;
+                case SHUNT_COMPENSATOR -> IdentifiableType.SHUNT_COMPENSATOR;
+                case DANGLING_LINE -> IdentifiableType.DANGLING_LINE;
+                case STATIC_VAR_COMPENSATOR -> IdentifiableType.STATIC_VAR_COMPENSATOR;
+                case VSC_CONVERTER_STATION, LCC_CONVERTER_STATION -> IdentifiableType.HVDC_CONVERTER_STATION;
```

Comment:
I was wondering how you make any difference between them then! but then I saw you get it from VscCount / LccCount method :+1: